### PR TITLE
[v3.6.0] Refactor native Batcher2d lifecycle

### PR DESCRIPTION
**Pull request by dumganhar**
-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by dumganhar** [comment]
@cocos-robot run test cases

File: cocos/2d/renderer/batcher-2d.ts [hunk 1]
```diff
@@ -123,11 +123,6 @@ export class Batcher2D implements IBatcher {
         this.device = _root.device;
         this._batches = new CachedArray(64);
         this._drawBatchPool = new Pool(() => new DrawBatch2D(), 128, (obj) => obj.destroy(this));
-
-        if (JSB) {
-            this._nativeObj = new NativeBatcher2d();
-            //this.initAttrBuffer();
-        }
```
Comment (minggo) [inline cocos/2d/renderer/batcher-2d.ts:130]: Why delete it? Then where to create the native object?

Comment (dumganhar) [reply cocos/2d/renderer/batcher-2d.ts:130]: NativeBatcher2d is allocated in Root::setRenderPipeline and JS visits it through Root::getBatcher2D jsb function.

File: native/cocos/2d/renderer/Batcher2d.cpp [hunk 2]
```diff
@@ -61,7 +61,7 @@ void Batcher2d::syncMeshBuffersToNative(uint32_t accId, ccstd::vector<UIMeshBuff
 }
 
 UIMeshBuffer* Batcher2d::getMeshBuffer(uint32_t accId, uint32_t bufferId) { // NOLINT(bugprone-easily-swappable-parameters)
-    auto map = _meshBuffersMap[accId];
+    const auto& map = _meshBuffersMap[accId];
```
Comment (dumganhar) [inline native/cocos/2d/renderer/Batcher2d.cpp:64]: Use `const auto&` to avoid memory allocation

**Review by minggo — APPROVED** [review]

**Review by SantyWang — APPROVED** [review]

**Review by Zhmz — APPROVED** [review]

**Pull request merged**